### PR TITLE
1 0 recording refactor plat 5808

### PR DESCRIPTION
**Pull request by ronyad**

### CODE REVIEW

File: recordingUploader/config.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+
+def config_section_default():
+    Config.read("config.ini")
+    section = "default_config"
+    fill(section)
+
+
+def get_config(key, type = lambda str: str):
+    if key in config_json:
```
Comment (kalturaguy) [inline recordingUploader/config.py]: check eval

File: recordingUploader/config.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        except Exception :
+            print("exception on %s!" % option)
+            config_json[option] = None
+
+
+def config_section_map():
+    Config.read("configMapping.ini")
+    config_sections = Config.sections()
+    for config_section in config_sections:
+        if config_section == hostname:
```
Comment (kalturaguy) [inline recordingUploader/config.py]: do regex instead

File: recordingUploader/TaskRunner.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    def move_and_add_to_queue(self, src_dir):
+        entry_regex = '^([01]_\w{8})_'
+        pattern = re.compile(entry_regex)
+        for directory_name in os.listdir(src_dir):
+            directory_path = os.path.join(src_dir, directory_name)
+            if os.path.isdir(directory_path) and pattern.match(directory_name) is not None:
+                try:
+                    if src_dir != self.working_directory:   # if its not the same directory
+                        shutil.move(directory_path, self.working_directory)
```
Comment (kalturaguy) [inline recordingUploader/TaskRunner.py]: check os.rename instead

File: recordingUploader/TaskRunner.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+            try:
+                src = os.path.join(self.working_directory, task_parameter['directory'])
+                job = self.task(task_parameter)  # operate the function task_job, with argument task_parameters
+                job.run()
+                shutil.move(src, self.output_directory)
+                self.logger.info("Task %s completed, Move %s to %s", self.task_name, src, self.output_directory)
+            except shutil.Error, e:
+                self.logger.error("Error while try to move directory into %s: %s",  self.output_directory, e.message)
+            except Exception as e:
+                self.logger.error("Error: %s", e.message)
```
Comment (kalturaguy) [inline recordingUploader/TaskRunner.py]: there is no callstack here

File: recordingUploader/TaskRunner.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        while True:
+            task_parameter = self.task_queue.get()
+            if self.task is None:   # todo checkit
+                break
+            self.logger.info("Task is performed by %d", index)
+            try:
+                src = os.path.join(self.working_directory, task_parameter['directory'])
+                job = self.task(task_parameter)  # operate the function task_job, with argument task_parameters
+                job.run()
+                shutil.move(src, self.output_directory)
```
Comment (kalturaguy) [inline recordingUploader/TaskRunner.py]: shutil.move

File: recordingUploader/TaskRunner.py [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+                    m = re.search(entry_regex, directory_name)
+                    entry_id = m.group(1)
+                    param = {'entry_id': entry_id, 'directory': directory_name}
+                    self.task_queue.put(param)
+                    self.logger.info("Add unhanded directory %s to the task queue", directory_name)
+                except Exception as e:
+                    self.logger.error(e)  # Catch error in case that the job is already taken by other machine
+
+    def work(self, index):
+        while True:
```
Comment (kalturaguy) [inline recordingUploader/TaskRunner.py]: how do you stop the thread?

File: recordingUploader/ServerUploader.py [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    def _generate_upload_thread(self):
+        for i in range(1, self.num_of_thread+1):
+            t = Thread(target=self.worker, args=(i,))  # todo change name
+            t.daemon = True  # todo check it
+            t.start()
+
+    def worker(self, index):
+        self.logger.info("Thread %d is start working", index)
+        while True:
+            item = self.q.get()
```
Comment (kalturaguy) [inline recordingUploader/ServerUploader.py]: add item.load()

File: recordingUploader/ServerUploader.py [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+
+    def upload_file(self, file_name):
+        try:
+            file_size = os.path.getsize(file_name)
+            infile = io.open(file_name, 'rb')
+            chunks_to_upload = int(file_size / self.upload_token_buffer_size) + 1
+            upload_session = self.kaltura_api.KalturaUploadSession(self.output_filename, file_size, chunks_to_upload, self.entry_id)
+
+            for sequence_number in range(1, chunks_to_upload+1):
+                data = infile.read(self.upload_token_buffer_size)
```
Comment (kalturaguy) [inline recordingUploader/ServerUploader.py]: move to item.load

File: recordingUploader/ServerUploader.py [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+                item = {
+                    'upload_session': upload_session,
+                    'data_stream': ImpersonateFile(file_name, data),
+                    'sequence_number': sequence_number,
+                    'is_last_chunk': sequence_number == chunks_to_upload,
+                    "resumeAt": self.upload_token_buffer_size * (sequence_number - 1),
+                    'resume': chunks_to_upload is not 1 and sequence_number is not 1
```
Comment (kalturaguy) [inline recordingUploader/ServerUploader.py]: why not sequence_number >0?

File: recordingUploader/ServerUploader.py [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+                    self.kaltura_api.set_media_content(item['upload_session'])
+            except IOError as e:  # todo which kind of excepotion? what to do then? verify that no more try is wrapped
+                self.logger.error("Failed to upload file: %s", e.message)
+            self.q.task_done()
+
+    def upload_file(self, file_name):
+        try:
+            file_size = os.path.getsize(file_name)
+            infile = io.open(file_name, 'rb')
+            chunks_to_upload = int(file_size / self.upload_token_buffer_size) + 1
```
Comment (kalturaguy) [inline recordingUploader/ServerUploader.py]: file_size could be equal to upload_token_buffer_size...

File: recordingUploader/ServerUploader.py [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+import Queue
+from threading import Thread
+import logging.handlers
+from TaskRunner import TaskBase
+import time
+from random import randint
+
+
+class ImpersonateFile: #
+    def __init__(self, file_name, _buffer):
```
Comment (kalturaguy) [inline recordingUploader/ServerUploader.py]: move to a different file

File: recordingUploader/KalturaAPI.py [hunk 12]
```diff
@@ -0,0 +1,451 @@
...
+        self.logger.debug("execution time for [%s]: [%s]" % (url, end_time - start_time))
+
+        # print server debug info to log
+        server_name = None
+        server_session = None
+        for curHeader in response_headers:
+            if curHeader.startswith('X-Me:'):
+                server_name = curHeader.split(':', 1)[1].strip()
+            elif curHeader.startswith('X-Kaltura-Session:'):
+                server_name = curHeader.split(':', 1)[1].strip()
```
Comment (kalturaguy) [inline recordingUploader/KalturaAPI.py]: server_session

File: recordingUploader/KalturaAPI.py [hunk 13]
```diff
@@ -0,0 +1,451 @@
...
+            self.file_size = file_size
+            self.chunks_to_upload = chunks_to_upload
+            self.partner_id = KalturaAPI().get_partner_id(entry_id)
+            self.upload_entry = KalturaAPI().create_entry(self.partner_id, "text", "text")
+            self.token_id = KalturaAPI().upload_token_add(self.partner_id, file_name, file_size)
+
+    @staticmethod
+    def append_recording(file_path):
+        shutil.move(file_path,
+                    '/Users/ron.yadgar/dvr/isilon')  # if any file are in procceing dir, move to archive
```
Comment (kalturaguy) [inline recordingUploader/KalturaAPI.py]: ?????

File: recordingUploader/TaskRunner.py (hunk 3 above)
Comment (ronyad) [inline recordingUploader/TaskRunner.py]: not work for directories

**Pull request merged**